baseapp/abci.go

```diff
@@ -1214,46 +1220,71 @@ func (app *BaseApp) CreateQueryContext(height int64, prove bool) (sdk.Context, e
 			)
 	}
 
-	// when a client did not provide a query height, manually inject the latest
-	if height == 0 {
-		height = lastBlockHeight
-	}
-
-	if height <= 1 && prove {
+	if height > 0 && height <= 1 && prove {
```

Comment:
doesnt this just equate to `if height == 1 && prove` ?

Reply:
yeah, however i think it should just be `height <= 1`?

Reply:
nvm that seems to have broken the tests